addons/hr_holidays/static/src/bugfix/bugfix_tests.js

```diff
@@ -16,3 +16,111 @@ QUnit.module('bugfix_tests.js', {
+         * @param {Object} [otherProps={}]
+         * @param {Object} [param2={}]
+         * @param {boolean} [param2.isFixedSize=false]
+         */
+        this.createThreadViewComponent = async (threadView, otherProps = {}, { isFixedSize = false } = {}) => {
+            let target;
+            if (isFixedSize) {
+                // needed to allow scrolling in some tests
+                const div = document.createElement('div');
+                Object.assign(div.style, {
+                    display: 'flex',
+                    'flex-flow': 'column',
+                    height: '300px',
+                });
+                this.widget.el.append(div);
+                target = div;
+            } else {
+                target = this.widget.el;
+            }
+            const props = Object.assign({ threadViewLocalId: threadView.localId }, otherProps);
+            await createRootComponent(this, components.ThreadView, { props, target });
+        };
+
+        this.start = async params => {
+            const { afterEvent, env, widget } = await start(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.afterEvent = afterEvent;
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        afterEach(this);
+    },
+});
+
+QUnit.test('out of office message on direct chat with out of office partner', async function (assert) {
+    assert.expect(2);
+
+    // Returning date of the out of office partner, simulates he'll be back in a month
+    const returningDate = moment.utc().add(1, 'month');
+    // Needed partner & user to allow simulation of message reception
+    this.data['res.partner'].records.push({
+        id: 11,
+        name: "Foreigner partner",
+        out_of_office_date_end: returningDate.format("YYYY-MM-DD HH:mm:ss"),
+    });
+    this.data['mail.channel'].records = [{
+        channel_type: 'chat',
+        id: 20,
+        members: [this.data.currentPartnerId, 11],
+    }];
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_outOfOffice',
+        "should have an out of office alert on thread view"
+    );
+    const formattedDate = returningDate.toDate().toLocaleDateString(
+        this.env.messaging.locale.language.replace(/_/g,'-'),
+        { day: 'numeric', month: 'short' }
+    );
+    assert.ok(
+        document.querySelector('.o_ThreadView_outOfOffice').textContent.includes(formattedDate),
+        "out of office message should mention the returning date"
+    );
```

Comment:
Won't this test fail in December due to going over next year?

Reply:
No, I tested it, the output will add the year but we only check that month and day are presents